spec/zeitwerk_compliance_spec.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'ZeitwerkCompliance' do
+  describe 'when eager loading' do
+    it 'loads all files without errors' do
+      expect { Zeitwerk::Loader.eager_load_all }.not_to raise_error
```

Comment:
For explicitness, maybe we could glue together the configuration require here?

```suggestion
      expect do
        require 'msfenv'
        Zeitwerk::Loader.eager_load_all
      end.not_to raise_error
```